Source/VMware.vSphereDSC/DSCResources/VMHostAdvancedSettings.ps1

```diff
@@ -0,0 +1,225 @@
+        if ($null -eq $advancedSetting) {
+            <#
+            Here 'Write-Warning' is used instead of 'throw' to ensure that the execution will not stop
+            if an invalid Advanced Setting is present in the passed hashtable and in the same time to
+            provide an information to the user that invalid data is passed.
+            #>
+            Write-Warning "Advanced Setting $advancedSettingName does not exist for VMHost $($vmHostName) and will be ignored."
+        }
+
+        return $advancedSetting
+    }
+
+    <#
+    .DESCRIPTION
+
+    Converts the desired value of the Advanced Setting from the hashtable to the correct type of the value
+    from the server. Works only for primitive types.
+    #>
+    [object] ConvertAdvancedSettingDesiredValueToCorrectType($advancedSetting) {
+        $advancedSettingDesiredValue = $null
+        if ($advancedSetting.Value -is [bool]) {
+            # For bool values the '-as' operator returns 'True' for both 'true' and 'false' strings so specific conversion is needed.
+            $advancedSettingDesiredValue = [System.Convert]::ToBoolean($this.AdvancedSettings[$advancedSetting.Name])
+        }
+        else {
+            $advancedSettingDesiredValue = $this.AdvancedSettings[$advancedSetting.Name] -as $advancedSetting.Value.GetType()
+        }
+
+        return $advancedSettingDesiredValue
+    }
+
+    <#
+    .DESCRIPTION
+
+    Checks if the Advanced Setting should be updated depending on the passed desired value.
+    #>
+    [bool] ShouldUpdateVMHostAdvancedSetting($advancedSetting) {
+        <#
+        Each element in the hashtable is of type MSFT_KeyValuePair where the value is a string.
+        So before comparison the value needs to be converted to its original type which can be
+        retrieved from its server value.
+        #>
+        $advancedSettingDesiredValue = $this.ConvertAdvancedSettingDesiredValueToCorrectType($advancedSetting)
+
+        return ($advancedSettingDesiredValue -ne $advancedSetting.Value)
+    }
+
+    <#
+    .DESCRIPTION
+
+    Checks if any of the Advanced Settings present in the hashtable need to be updated.
+    #>
+    [bool] ShouldUpdateVMHostAdvancedSettings($vmHost) {
+        $retrievedAdvancedSettings = $this.GetAdvancedSettings($vmHost)
+
+        foreach ($advancedSettingName in $this.AdvancedSettings.Keys) {
+            $advancedSetting = $this.GetAdvancedSetting($retrievedAdvancedSettings, $advancedSettingName, $vmHost.Name)
+
+            if ($null -ne $advancedSetting -and $this.ShouldUpdateVMHostAdvancedSetting($advancedSetting)) {
+                return $true
+            }
+        }
+
+        return $false
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns the Option Manager of the specified VMHost.
+    #>
+    [PSObject] GetOptionManager($vmHost) {
+        try {
+            $optionManager = Get-View -Server $this.Connection -Id $vmHost.ExtensionData.ConfigManager.AdvancedOption -ErrorAction Stop
```

Comment:
Why settings are updated through views, couldn't you use the cmdlets?

Reply:
I used this approach because if Set-AdvancedSetting is used, for each advanced setting a call to the server needs to be made and with the current approach only one call to the server is made to update all passed Advanced Settings.